spec/unit/provider/elasticsearch_shield_role/parsed_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'spec_helper'
+
+describe Puppet::Type.type(:elasticsearch_shield_role).provider(:parsed) do
+
+  on_supported_os.each do |os, facts|
+
+    let(:facts) { facts }
+
+    describe 'instances' do
+      it 'should have an instance method' do
+        expect(described_class).to respond_to :instances
+      end
+
+      context 'with no roles' do
+        it 'should return no resources' do
+          expect(described_class.parse("\n")).to eq([])
+        end
+      end
+
+      context 'with one role' do
+        it 'should return one resource' do
+          expect(described_class.parse(%q{
+admin:
```

Comment:
Could I persuade to indent these heredocs? If you care about having no leading whitespace, you could use a method on the heredoc, like this one using the (trivial) `unindent` Gem.

```
[19] pry(main)> heredoc = <<-EOF.unindent
[19] pry(main)*   stuff:
[19] pry(main)*     - more
[19] pry(main)*     - stuff
[19] pry(main)*   EOF
=> "stuff:\n  - more\n  - stuff\n"

[20] pry(main)> puts heredoc
stuff:
  - more
  - stuff
=> nil
```